opensir/models/model.py

```diff
@@ -74,10 +80,10 @@ def set_params(self, p, initial_conds):
         params = self.PARAMS
         ic = self.IC
 
-        if type(p) is dict:
+        if isinstance(p, dict):
```

Comment:
Strictly speaking we are only allowing dicts here. So, `isinstance` would also consider objects that inherit from dicts.

Reply:
well, pylint complains there. So, let's leave it as "isinstance" :)